File: RedSheet.DbEntities/Models/Role.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("Roles", Schema = "dbo")]
	[Serializable]
	public partial class Role  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int RoleId { get; set; } 
 
	[Required] 
	[MaxLength(50)]
	public string RoleName { get; set; } 
 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	public Status Status { get; set; } 
	[InverseProperty("Role")]
	public ICollection<RolePermission> RolePermissions { get; set; } 


	public Role()
	{
	this.RolePermissions = new HashSet<RolePermission>();

    }

    }
}

File: redsheet-portal/src/app/database-models/role.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { RolePermission,  } from './'
export class Role {
    constructor(role?: Role )  {
        let properties = [ "roleId", "roleName", "status", "rolePermissions",];
        for (let property of properties)
            if (role && role[property])
                this[property] = role[property];
    }
 
	roleId : number =   0 ;
 
    @required()
    @maxLength(50)
	roleName : string =   undefined;
 
    @range(0,2147483647)
	status : number =   undefined;
	@nested(RolePermission)
	rolePermissions: RolePermission[];



}
